rcl/src/rcl/timer.c

```diff
@@ -294,15 +294,15 @@ rcl_timer_call(rcl_timer_t * timer)
   // between the timer being ready and the callback being triggered
   next_call_time += period;
   // in case the timer has missed at least once cycle
-  if (next_call_time < now) {
+  if (next_call_time <= now) {
     if (0 == period) {
       // a timer with a period of zero is considered always ready
       next_call_time = now;
```

Comment:
period of zero case is handled here.

Reply:
I only changed the behaviour for periods greater zero. With zero it will get called every tick, even if the sim clock did not advance. But that may be expected. Not expected is, that a timer with a period gets called twice without the clock advancing.